examples/interactionsDemo.html

```diff
@@ -0,0 +1,517 @@
+                if (!redirectUrl) {
+                    redirectUrl = window.location.protocol + window.location.host;
+                }
+                redirectUrl += window.location.pathname;
+
+                let reloadingEl = document.querySelector('.reloading');
+
+                /**
+                 *  For testing purposes using the text box to manually find a conversation by ID,
+                 *  does not care about permissions to view interactions.
+                 */
+                $('#submitConversationIDButton').on('click', function() {
+                    let conversationId = $('#conversationID').val().trim();
+                    clientApp.conversations.showInteractionDetails(conversationId);
+                });
+
+                // Authenticate with PureCloud
+                let authenticatingEl = document.querySelector('.authenticating');
+                let userDetailsEl = document.querySelector('.userDetails');
+
+                setHidden(authenticatingEl, false);
+
+                let authenticated = false;
+                let canViewInteraction = false;
+                let canViewWrapUpCodes = false;
+
+                client.loginImplicitGrant(clientId, redirectUrl, { state: ('pcEnvironment=' + pcEnvironment) })
+                    .then(function () {
+                        authenticated = true;
+                        setHidden(userDetailsEl, false);
+                        return new platformClient.UsersApi().getUsersMe({ 'expand': ['authorization'] });
+                    })
+                    .then(function (profileData) {
+                        $('#name').append(' ' + profileData.name);
+
+                        // Check if the current user will be able to even view the interaction before making the request
+                        let permissions = profileData.authorization.permissions;
+                        if (checkPermission(permissions, 'analytics:conversationDetail:view') &&
+                            (checkPermission(permissions, 'conversation:communication:view') ||
+                             checkPermission(permissions, 'quality:evaluation:add') ||
+                             checkPermission(permissions, 'quality:calibration:view') ||
+                             checkPermission(permissions, 'quality:evaluation:editScore'))) {
+                                canViewInteraction = true;
+                                $('#role').append(' Supervisor');
+                        } else {
+                                $('#role').append(' Agent');
+                        }
+
+                        if (checkPermission(permissions, 'routing:wrapupCode:view')) {
+                            canViewWrapUpCodes = true;
+                        }
+
+                        setHidden(authenticatingEl, true);
+
+                        $('#reloadButton').on('click', function () {
+                            $('.failure').text('');
```

Comment:
If I understand what you are saying about the error state correctly, does this line and line 228 handle all cases correctly by hiding the failure message and removing the table body? This would be triggered on every page load and button reload.

Reply:
Yeah, I think only adding the click listener after successful auth and user retrieval limits some of the edge cases I was seeing before.  Now, the error text will only be used for reloads after initial success, so there shouldn't be any polluting state.

We're definitely pushing the limits of what we can do in these demo apps by keeping it "simple" with no build, dependencies, etc.  Reminds me of the good 'ol days of jquery spaghetti!